app/js/controller/list_controller.js

```diff
@@ -165,20 +165,27 @@ export default class ListController extends Controller {
     }
 
     installReq.onerror = (err) => {
+      var errorMsg = err.target.error.name;
       var errorType = err.target.error.name;
       switch(errorType) {
         case 'DENIED':
           // If the user cancelled the install, we do nothing.
-          break;
+          return;
 
         case 'NETWORK_ERROR':
-          this.showAlertDialog('Install Error: No network');
+          errorMsg = 'No network';
+          break;
+
+        case 'MANIFEST_URL_ERROR':
+          errorMsg = 'Invalid manifest url';
           break;
 
-        default:
-          this.showAlertDialog('Error installing: ' + err.target.error.name);
+        case 'INVALID_SECURITY_LEVEL':
+          errorMsg = 'Invalid permissions';
```

Comment:
We should probably display something about enabling developer mode here. That can be a follow-up, though.